### ISSUE TITLE
Wrong "Create Selectable SP" output

### ISSUE THREAD

**Issue by arvanus** [mariuz/flamerobin#78]
(Same bug for EXECUTE BLOCK statement)
![image](https://user-images.githubusercontent.com/8182958/91339608-1b346b00-e7ad-11ea-8836-48f6e031bbca.png)

![image](https://user-images.githubusercontent.com/8182958/91339508-f8a25200-e7ac-11ea-881c-e2597bdf89f4.png)
![image](https://user-images.githubusercontent.com/8182958/91339644-27202d00-e7ad-11ea-91da-8b2f9f23a6e7.png)
Output:
![image](https://user-images.githubusercontent.com/8182958/91339702-3a32fd00-e7ad-11ea-9508-b8d14ed9e24a.png)
If I change in View->Preferences->Fields to "Only show datatype" it creates correctly:
![image](https://user-images.githubusercontent.com/8182958/91339883-82521f80-e7ad-11ea-86aa-4076e34212af.png)
(but ideally would be to show the domain)
![image](https://user-images.githubusercontent.com/8182958/91339968-9d249400-e7ad-11ea-8f66-59eea07e152a.png)
With show both domain and datatype, appears like this (bug):
![image](https://user-images.githubusercontent.com/8182958/91340073-c6ddbb00-e7ad-11ea-8560-530fc97b1a96.png)

I think that the same routine is used both for initial TreeView and for this, I think that when extracting FR would need to use the field domain when existing and the datatype elsewhere.
I would like to get some suggestions of what would be the most "elegant" way to give to the user this SQL

Thanks

**Comment by Jdochoa** [comment]
I created two new types of parameters 
- Source 
- TypeOf. 
Please see selectable procedure for a example of "Type Of" and selectable Execute Block for a example of "Source"

**Comment by luronumen** [comment]
Hi @arvanus,
 
Is this issue still reproducible in [0.9.3.8 Snapshot](https://github.com/mariuz/flamerobin/releases/tag/0.9.3.8) version?

**Comment by arvanus** [comment]
Hi, almost there!
At "Create selectable stored procedure", it's correctly using "type of"
At "EXECUTE BLOCK", it's always using the domain, even when there is no domain so it's using RDB$##, needs to fix this:
![image](https://user-images.githubusercontent.com/8182958/121020656-f24af880-c776-11eb-9cc4-651a2411be0b.png)

Note: `type of column` was add at Firebird 2.5, so this might be optional 
https://firebirdsql.org/refdocs/langrefupd25-psql-declare.html

**Comment by Jdochoa** [comment]
> 
> 
> Hi, almost there!
> At "Create selectable stored procedure", it's correctly using "type of"
> At "EXECUTE BLOCK", it's always using the domain, even when there is no domain so it's using RDB$##, needs to fix this:
> ![image](https://user-images.githubusercontent.com/8182958/121020656-f24af880-c776-11eb-9cc4-651a2411be0b.png)
> 
> Note: `type of column` was add at Firebird 2.5, so this might be optional
> https://firebirdsql.org/refdocs/langrefupd25-psql-declare.html

Hi @arvanus 

I created two new types of parameters

    Source
    TypeOf.
    Please see selectable procedure for a example of "Type Of" and selectable Execute Block for a example of "Source"

if you need execute block with "Type Of" please change the your local file create_selectable_execute_block

./jo

**Comment by arvanus** [comment]
Yes, but where it's not used a domain, Fr needs to deliver the primary type, not the "RDB$" domain

**Comment by mariuz** [comment]
Fixed by https://github.com/mariuz/flamerobin/pull/521
